### PR TITLE
[14.0][IMP] rma_account: add default journal

### PR DESCRIPTION
**Pull request by MateuGForgeFlow**
Add default journal in rma.operation, if the field is filled it will be used the selected journal, otherwise the module will have the normal flow.
cc @LoisRForgeFlow @AdriaGForgeFlow @JordiBForgeFlow

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@MateuGForgeFlow Can you add a test to make sure it works as designed?

File: rma_account/models/rma_operation.py [hunk 1]
```diff
@@ -17,3 +17,26 @@ class RmaOperation(models.Model):
...
         default="no",
     )
+
+    refund_journal_id = fields.Many2one(
+        comodel_name="account.journal",
+        string="Refund Account Journal",
+        domain="[('id', 'in', valid_journal_ids)]",
+    )
+
+    valid_journal_ids = fields.One2many(
```
Comment (LoisRForgeFlow) [inline rma_account/models/rma_operation.py]: valid_refund_journal_ids

File: rma_account/wizards/rma_refund.py [hunk 2]
```diff
@@ -128,14 +128,17 @@ def prepare_refund_line(self, item):
...
-            journal = self.env["account.journal"].search(
-                [("type", "=", "sale")], limit=1
-            )
+        if rma_line.operation_id.refund_journal_id:
+            journal = rma_line.operation_id.refund_journal_id
         else:
-            journal = self.env["account.journal"].search(
-                [("type", "=", "purchase")], limit=1
-            )
+            if rma_line.type == "customer":
```
Comment (LoisRForgeFlow) [inline rma_account/wizards/rma_refund.py]: elif and remove above else for more clean construction

**Review by LoisRForgeFlow — COMMENTED** [review]
Small details, LGTM

File: rma_account/models/rma_operation.py (hunk 1 above)
Comment (LoisRForgeFlow) [reply rma_account/models/rma_operation.py]: Also this should be a Many2many, it is compute and becouse of that doesn't fail, but strictly speaking this is many2many.

**Review by AdriaGForgeFlow — APPROVED** [review]

**Comment by JordiBForgeFlow** [comment]
@MateuGForgeFlow pre-commit is failing

File: rma_account/tests/test_rma_account.py [hunk 3]
```diff
@@ -26,11 +26,25 @@ def setUpClass(cls):
...
+            }
+        )
+
         # Create RMA group and operation:
+        cls.rma_order_customer = cls.rma_obj.create(
+            {"partner_id": customer1.id, "type": "customer"}
+        )
         cls.rma_group_customer = cls.rma_obj.create(
             {"partner_id": customer1.id, "type": "customer"}
         )
```
Comment (LoisRForgeFlow) [inline rma_account/tests/test_rma_account.py]: you are duplicating this, isn't it?

**Review by LoisRForgeFlow — APPROVED** [review]

File: rma_account/models/rma_operation.py [hunk 4]
```diff
@@ -17,3 +17,26 @@ class RmaOperation(models.Model):
...
+
+    valid_refund_journal_ids = fields.Many2many(
+        comodel_name="account.journal",
+        compute="_compute_domain_valid_journal",
+    )
+
+    @api.onchange("type")
+    def _compute_domain_valid_journal(self):
+        for rec in self:
+            if self.type == "customer":
```
Comment (bodedra) [inline rma_account/models/rma_operation.py]:
```suggestion
            if rec.type == "customer":
```

**Review by bodedra — COMMENTED** [review]
Minor improvement. Remaining LG :+1:

File: rma_account/models/rma_operation.py (hunk 4 above)
Comment (MateuGForgeFlow) [reply rma_account/models/rma_operation.py]: You are right! Thank you @bodedra

**Review by JordiBForgeFlow — APPROVED** [review]

**Comment by JordiBForgeFlow** [comment]
@MateuGForgeFlow Can you backport the changes to 13.0 and 12.0?

**Pull request merged**